src/main/java/com/kainos/ea/dao/UserDao.java

```diff
@@ -0,0 +1,58 @@
+package com.kainos.ea.dao;
+
+import com.kainos.ea.exception.UserAlreadyExistsException;
+import com.kainos.ea.model.User;
+import java.sql.*;
+public class UserDao {
+
+    public User registerUser(Connection conn, User user) throws SQLException {
+
+        User returnedUser = getUser(conn, user);
+
+        if (returnedUser.getId() != 0) {
+            User nullUser = new User();
```

Comment:
Is this when the user already exists? It should raise an error/exception rather than returning an empty user